### PR TITLE
SoA fixes (2nd round): fix and add broken / missing SoA (const) row-to-row assignment.

### PR DESCRIPTION
**Pull request by ericcano**
#### PR description:

This PR adds a unit test validating the row to row copy in SoA views. It fixed the mutable row to mutable row assignment and adds as const_element (row) to element (row) assignemt.
Minor documentation fix.

#### PR validation:

A unit test was created to validate the functionalities. No regression is expected as the functionality was not usable.

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: DataFormats/SoATemplate/test/SoAUnitTests.cc
Comment (fwyzard) [inline DataFormats/SoATemplate/test/SoAUnitTests.cc:1]: @ericcano thanks for the new test :-)

Could you add a comment here in the PR about which of the assignments in the test were broken without this PR ?

File: DataFormats/SoATemplate/test/SoAUnitTests.cc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    SimpleLayout::ConstView slcv{sl};
+    auto slv0 = slv[0];
+    slv0.x() = 1;
+    slv0.y() = 2;
+    slv0.z() = 3;
+    slv0.t() = 5;
+    // Fill up
+    for (SimpleLayout::View::size_type i = 1; i < slv.metadata().size(); ++i) {
+      auto slvi = slv[i];
+      slvi = slv[i - 1];
```
Comment (ericcano) [inline DataFormats/SoATemplate/test/SoAUnitTests.cc:35]: This assignment was failing without the change in `SoACommon.h` (bug in the const flavor of the mutable row accessor, triggered by the fact that the operator=() parameter is const).

File: DataFormats/SoATemplate/test/SoAUnitTests.cc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        SimpleLayout::ConstViewTemplate<cms::soa::RestrictQualify::disabled, cms::soa::RangeChecking::Default>;
+    View slv{sl};
+    ConstView slcv{sl};
+    auto slv0 = slv[0];
+    slv0 = {7, 11, 13, 17};
+    // Fill up
+    for (SimpleLayout::View::size_type i = 1; i < slv.metadata().size(); ++i) {
+      auto slvi = slv[i];
+      // Copy the const view
+      slvi = slcv[i - 1];
```
Comment (ericcano) [inline DataFormats/SoATemplate/test/SoAUnitTests.cc:76]: This assignement was missing (added to `SoAView.h`)

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by rappoccio** [comment]
+1

**Pull request merged**